### ISSUE TITLE
Custom TS loaders are not working

### ISSUE THREAD

**Issue by taletski** [dotansimha/graphql-code-generator#8981]
### Which packages are impacted by your issue?

@graphql-codegen/cli

### Describe the bug

I am following the [guide](https://the-guild.dev/graphql/codegen/docs/config-reference/schema-field#loading-api-url-from-typescript-file-example) for creating a custom loader using TS. 

It appears that the codegen fails to load the `.ts` loader file yet works ok with a `.js` one.

### Your Example Website or App

https://codesandbox.io/p/sandbox/graphql-codegen-custom-schema-loader-issue-hkve7f

### Steps to Reproduce the Bug or Issue

- Run `yarn generate`
The command fails with an error

- Change `loader.ts` to `loader.js` in `codegen.ts`
- Run `yarn generate`
This time the command will work

### Expected behavior

As a user, I expected that the custom schema loader written in TS will work if I follow the [guide](https://the-guild.dev/graphql/codegen/docs/config-reference/schema-field#loading-api-url-from-typescript-file-example)

### Platform

- OS: [e.g. macOS, Linux]
- NodeJS: 16.17.0
- `graphql` version: 16.2.0
- `@graphql-codegen/*` version(s): 2.6.2

### Codegen Config File

import { CodegenConfig } from "@graphql-codegen/cli";

const config: CodegenConfig = {
  schema: {
    "schema.graphql": {
      // loader: "./loader.js", // JS - ✅ working
      loader: "./loader.ts", // TS - ❌ not working
    },
  },
  documents: "document.graphql",
  generates: {
    "types.ts": { plugins: ["typescript", "typescript-operations"] },
  },
};

export default config;

### Additional context

Please, note that I am registering the `ts-node` for the codegen command.
```
graphql-codegen -r ts-node/register --config ./codegen.ts
```

**Comment by Axel-P** [comment]
Just a note on reproduction details. This is happening to us explicitly on `3.2.2`
We have this condegen config:
```
const config = {
  schema: {
    [schemaUrl]: {
      headers: {
        Authorization: AUTHORIZATION
      }
    }
  },
  documents: {
    'path/to/my/query': {
      templateValues: 'my/template/values.json',
      loader: './documentLoader.ts'
    }
  },
  config: {
    skipTypeName: true
  },
  generates: {
    ....
  }
}
```
Which runs just fine on codegen `3.2.1` and lower, but we started a new project where we installed `3.2.2` and it fails not able to find the loader:
![Screenshot 2023-03-08 at 11 18 20 AM](https://user-images.githubusercontent.com/11817345/223769036-5c6c5a13-bc0c-4411-b7c4-9689a694197f.png)
On the initial run of `generate:local` package.json was set to install codegen `^3.2.1` which would make npm fetch the `3.2.2` version. When forcing npm to install `3.2.1` the command ran without issues (yes, we only had a single `^3.2.1` package, so `sed` worked just fine)

**Comment by dottodot** [comment]
Yes I'm seeing the same 3.2.1 works fine but upgrading to 3.2.2 gives an error of  Failed to load custom loader
